source/read-write-pref.txt

```diff
@@ -0,0 +1,237 @@
+detail on the ``read_concern`` type and definitions of the read concern levels, see
+`Read Concern <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Client.html#read_concern-instance_method>`__ 
+in the API documentation.
+
+Example: Set the Read Concern Level of an Aggregation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following code sets the read concern level of an aggregation to ``"available"``:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-read-concern
+   :end-before: end-read-concern
+
+For more information about aggregation, see the Aggregation page.
+
+.. TODO: insert :ref:`Aggregation <ruby-aggregation>` link here
+
+Example: Change the Read Concern of a Database
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following code changes the read concern level of a database to ``"local"``:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-change-read-concern
+   :end-before: end-change-read-concern
+
+Read Preference
+---------------
+
+The read preference determines which member of a replica set MongoDB reads when running a
+query.
+
+For more detailed API documentation, see the `Read Preference API
+documentation <https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Collection/View/Readable.html#read_preference-instance_method>`__.
+
+The following table shows options you can use to customize how the server evaluates 
+members:
+
+.. list-table::
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Parameter
+     - Type
+     - Description
+
+   * - ``mode``
+     - ``Symbol``
+     - Specifies a requirement or preference for which replica set
+       member the server reads from. The default mode, ``:primary``, specifies that
+       operations read from the primary member of the replica set.
+
+   * - ``tags`` *(optional)*
+     - ``Array<Hash>``
+     - Assigns tags to secondary replica set members to customize how the server evaluates
+       them. Tags cannot be used with the ``:primary`` read preference mode setting.
+
+   * - ``options`` *(optional)*
+     - ``Hash``
+     - Sets various options, including :manual:`hedge </core/read-preference-hedge-option/>`
+       and :manual:`maxStalenessSeconds </core/read-preference-staleness/>` that can be
+       applied to your read preference.
+
+Example: Set Read Preference and Concerns for a Transaction
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following code sets the read preference, read concern, and write concern for
+the operations in a transaction:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-read-preference
+   :end-before: end-read-preference
+
+For more information about transactions, see the Transactions guide.
```

Comment:
I think this should be commented out as well until the TODO is done, that way it doesn't look like a broken link (applies to all TODO links)